test/test_modules.py

```diff
@@ -734,6 +736,35 @@ def test_if_train_and_eval_modes_differ(self, device, dtype, module_info, traini
                     raise e
 
 
+    @onlyCPU
+    @modules(module_db)
+    def test_device_ctx_init(self, device, dtype, module_info, training):
+        module_cls = module_info.module_cls
+        module_inputs = module_info.module_inputs_func(module_info, device=device, dtype=dtype,
+                                                       requires_grad=False, training=training)
+        module_inputs_meta = module_info.module_inputs_func(module_info, device='meta', dtype=dtype,
+                                                            requires_grad=False, training=training)
+
+        for module_input, module_input_meta in zip(module_inputs, module_inputs_meta):
+            c_args, c_kwargs = module_input.constructor_input.args, module_input.constructor_input.kwargs
+            fw_args, fw_kwargs = module_input.forward_input.args, module_input.forward_input.kwargs
+
+            c_args_meta, c_kwargs_meta = module_input_meta.constructor_input.args, module_input_meta.constructor_input.kwargs
+            fw_args_meta, fw_kwargs_meta = module_input_meta.forward_input.args, module_input_meta.forward_input.kwargs
+
+            m_cpu = module_cls(*c_args, **c_kwargs)
+
+            with torch.device('meta'):
+                m = module_cls(*c_args_meta, **c_kwargs_meta)
```

Comment:
Why do you need to pass special meta inputs here?

Reply:
some of the classes (e.g. TransformerEncoderLayer, take in layers, so we need to make sure that the layers were are on meta)

Reply:
Should we create the args within the meta context then?

Reply:
Hm since the `module_inputs_func` on 745 that generates the list of module_inputs that hold the args takes a device kwarg I think that has roughly the same effect as using the meta device context manager to create the args

Reply:
Done above!